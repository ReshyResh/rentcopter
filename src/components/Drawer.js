import * as React from 'react';
import Box from '@mui/material/Box';
import Drawer from '@mui/material/Drawer';
import List from '@mui/material/List';
import { FaBars } from 'react-icons/fa';
import { NavLink } from 'react-router-dom';
import LogoutButton from './LogoutButton';
import Logo from './Logo';
import NavImages from './NavImages';

export default function TemporaryDrawer() {
  const [state, setState] = React.useState({
    left: false,
  });
  const isAdmin = sessionStorage.getItem('admin');
  const normalMenu = ['Reservations', 'Add reservation'];
  const adminMenu = ['Reservations', 'Add reservation', 'Add helicopter', 'Remove Helicopter'];
  const getMenu = () => (isAdmin === 'true' ? adminMenu : normalMenu);

  const toggleDrawer = (anchor, open) => (event) => {
    if (event.type === 'keydown' && (event.key === 'Tab' || event.key === 'Shift')) {
      return;
    }

    setState({ ...state, [anchor]: open });
  };

  const list = (anchor) => (
    <Box
      sx={{ width: anchor === 'top' || anchor === 'bottom' ? 'auto' : 250 }}
      role="presentation"
      onClick={toggleDrawer(anchor, false)}
      onKeyDown={toggleDrawer(anchor, false)}
      style={{ backgroundColor: '#1b1b13' }}
      className=" h-full text-white flex flex-col justify-center text-center md:hidden "
    >
      <List className="h-full flex flex-col justify-center">
        {getMenu().map((text) => (
          <div key={text} className="text-center border-b border-gray-500 border-opacity-70 w-full hover:shadow-yellow-400 hover:shadow-inner flex justify-evenly duration-500">
            <NavImages key={text} text={text} />
            <NavLink
              key={text}
              className="menu-items text-xl mb-2 transition-all p-5 w-3/4 "
              to={`/${text.toLowerCase().replace(' ', '-')}`}
            >
              {text}
            </NavLink>
          </div>

        ))}
        <Logo />
        <LogoutButton />
      </List>
    </Box>
  );

  return (
    <div className="sm:block md:hidden text-yellow-400 text-5xl absolute top-4 left-4 z-50">
      <FaBars onClick={toggleDrawer('left', true)} />
      <Drawer
        className="rounded-r-xl"
        anchor="left"
        open={state.left}
        onClose={toggleDrawer('left', false)}
      >
        {list('left')}
      </Drawer>
    </div>
  );
}
