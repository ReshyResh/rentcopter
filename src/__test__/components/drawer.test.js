import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Drawer from '../../components/Drawer';

it('Drawer component renders correctly', () => {
  const tree = renderer
    .create(<Router><Drawer /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
