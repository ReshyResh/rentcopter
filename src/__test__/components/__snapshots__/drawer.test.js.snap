// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer component renders correctly 1`] = `
<div
  className="sm:block md:hidden text-yellow-400 text-5xl absolute top-4 left-4 z-50"
>
  <svg
    fill="currentColor"
    height="1em"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
    />
  </svg>
</div>
`;
